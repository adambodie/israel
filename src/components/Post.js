import React, { Component } from 'react';
import { Row, Column } from 'react-foundation';
import ReactMarkdown from 'react-markdown';
import axios from 'axios';
import PhotoList from './PhotoList';

export default class Post extends Component {

	constructor() {
		super();
		this.state = {
			photographs: []
		};
	}
	componentDidMount() {
		axios.get(this.props.data)
			.then(response => {
				this.setState({
					photographs: response.data
				});
			})
			.catch(error => {
				console.log('Error fetching and parsing data', error);
			});
	}
	render() {
		return (
			<Row className="display grid-x post">
				<Column large={8} offsetOnLarge={2}>
					<h1>{this.props.title}</h1>
					<PhotoList data={this.state.photographs} />
					<ReactMarkdown source={this.props.paragraph} />
				</Column>
			</Row>
	)
}
};
