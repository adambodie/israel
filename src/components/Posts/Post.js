import React, { Component } from 'react';
import { Row, Column } from 'react-foundation';
import ReactMarkdown from 'react-markdown';
import axios from 'axios';
import PhotoList from './PhotoList';
import Quiz from '../Quiz/Quiz';

export default class Post extends Component {

	constructor() {
		super();
		this.state = {
			photographs: []
		};
	}
	componentDidMount() {
		axios.get(this.props.data)
			.then(response => {
				this.setState({
					photographs: response.data
				});
			})
			.catch(error => {
				console.log('Error fetching and parsing data', error);
			});
	}
	render() {
		return (
			<Row className="display grid-x post">
				<Column large={8} offsetOnLarge={2}>
					<h1>{this.props.title}</h1>
					<PhotoList data={this.state.photographs} />
					<ReactMarkdown source={this.props.paragraph} className="article" />
					{this.props.begin >= 0 && (<Quiz begin={this.props.begin} end={this.props.end} number={this.props.number}/>)}
				</Column>
			</Row>
	)
}
};
