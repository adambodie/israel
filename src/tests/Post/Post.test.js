import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Post from '../../components/Posts/Post';
import sinon from 'sinon';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';

beforeEach(() => {
	const mock = new MockAdapter(axios);
	const mockData = {data: {title: 'Picture', id:'4000'}};
	mock.onGet(mockData).reply(200, mockData)
});

it('renders correctly', () => {
	const mockData = {data: {title: 'Picture', id:'4000'}};
	const tree = renderer.create(
		<MemoryRouter>
			<Post 
				title='Title'
				paragraph={'Lorem ipsum dolor sit amet'} 
				prev={'prev'} 
				next={'next'}
				data={mockData}
				/>
		</MemoryRouter>
		).toJSON();
	expect(tree).toMatchSnapshot();
});

it('renders a `title`', () => {
	const wrapper = shallow(<Post title='Title' 
								paragraph={'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'} 
								prev={'prev'} 
								next={'next'} 
								/>
							);
	expect(wrapper.find('h1').length).toEqual(1);
 });

 it('calls componentDidMount', () => {
	const mockData = {data: {title: 'Picture', id:'4000'}};
    sinon.spy(Post.prototype, 'componentDidMount');
    const wrapper = mount(<MemoryRouter>
		<Post
			title='Title' 
			paragraph={'Lorem ipsum dolor sit amet, consectetur adipiscing elit.'} 
			prev={'prev'} 
			next={'next'}
			data={mockData}
			/>
	</MemoryRouter>);
    expect(Post.prototype.componentDidMount).toHaveProperty('callCount', 1);
  });


