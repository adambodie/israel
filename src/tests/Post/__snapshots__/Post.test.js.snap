// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="row display grid-x post"
>
  <div
    className="columns large-8 large-offset-2"
  >
    <h1>
      Title
    </h1>
    <div
      className="slick-slider slick-initialized"
    >
      
      <div
        className="slick-list"
      >
        <div
          className="slick-track"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          style={
            Object {
              "WebkitTransform": "translate3d(0px, 0px, 0px)",
              "WebkitTransition": "",
              "msTransform": "translateX(0px)",
              "opacity": 1,
              "transform": "translate3d(0px, 0px, 0px)",
              "transition": "",
            }
          }
        />
      </div>
      
      
    </div>
    <div
      className="article"
    >
      <p>
        Lorem ipsum dolor sit amet
      </p>
    </div>
    <div
      className="prev-next"
    >
      <a
        className="custom-button"
        href="/prev"
        onClick={[Function]}
      >
        Prev
      </a>
      <a
        className="custom-button"
        href="/next"
        onClick={[Function]}
      >
        Next
      </a>
    </div>
  </div>
</div>
`;
